src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -2,34 +2,123 @@
 
 import frc.robot.Const;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj.DriverStation;
 import edu.wpi.first.wpilibj.I2C;
+
+import com.revrobotics.CANPIDController;
 import com.revrobotics.CANSparkMax;
 import com.revrobotics.CANSparkMaxLowLevel.MotorType;
 import com.ctre.phoenix.motorcontrol.can.WPI_VictorSPX;
 
 import com.revrobotics.ColorSensorV3;
+import com.revrobotics.ControlType;
 
 /**
  * The Elevator subsystem controls the elevator mechanism
  *
- * This subsystem consists of the following components:
- * - The elevator motor (1x Spark Max using internal NEO encoder.)
- * - The adjustment wheel (1x Talon SRX/Victor SPX controller on either CAN or PWM)
- * - The color sensor for the color wheel
+ * This subsystem consists of the following components: - The elevator motor (1x
+ * Spark Max using internal NEO encoder.) - The adjustment wheel (1x Talon
+ * SRX/Victor SPX controller on either CAN or PWM) - The color sensor for the
+ * color wheel
  *
- * This subsystem should provide the following functions:
- * - Run a position loop on the elevator 
- * - Run a speed loop on the elevator
- * - Run a speed loop on the adjustment wheel
- * - Run a position loop for the color wheel (both matching color and rotations)
+ * This subsystem should provide the following functions: - Run a position loop
+ * on the elevator - Run a speed loop on the elevator - Run a speed loop on the
+ * adjustment wheel - Run a position loop for the color wheel (both matching
+ * color and rotations)
  */
 public class Elevator extends SubsystemBase {
 
     private final CANSparkMax elevatorMotor = new CANSparkMax(Const.CAN.ELEVATOR_MOTOR, MotorType.kBrushless);
     private final WPI_VictorSPX wheelMotor = new WPI_VictorSPX(Const.CAN.ELEVATOR_WHEEL);
     private final ColorSensorV3 colorSensor = new ColorSensorV3(I2C.Port.kOnboard);
+    private final CANPIDController elevatorPID = new CANPIDController(elevatorMotor);
+    private String targetColor = "";
+    private int revCount = 0;
+    private boolean done = false;
```

Comment:
Is this variable really required? Will the elevator ever really be done?

Reply:
What variable? There's a good handful in this highlight. The Boolean "done" was for the color wheel function that we already established was garbage, so this was garbage as well. I can't explain my logic behind this because I was working on flawed assumptions anyways.

Reply:
You're right, Github made it unclear. It's supposed to be the `done` variable. This is useless, because when the color wheel is done... well then the motor just stops moving. No need for a flag. Unless... I think it would be a good idea to return the state of the color wheel from the `spinColorWheel()` function... that is, return true if it's where it needs to be, and return false if it isn't. Then in our command, our `isFinished()` block can just look like this:

```java
@Override
public boolean isFinished() {
    return elevator.spinColorWheel(color);
}